tests/test_job.py

```diff
@@ -1,7 +1,141 @@
-from pasqal_cloud import Job
+from datetime import datetime
+from typing import Any, Generator, List, Union
+from unittest.mock import patch
+from uuid import UUID, uuid4
+
+import pytest
+
+from pasqal_cloud import (
+    Job,
+    JobCancellingError,
+    JobFetchingError,
+    JobFilters,
+    PaginatedResponse,
+    PaginationParams,
+    SDK,
+)
+from pasqal_cloud.utils.constants import JobStatus
+from tests.test_doubles.authentication import FakeAuth0AuthenticationSuccess
+from tests.utils import build_query_params
 
 
 class TestJob:
+    @pytest.fixture(autouse=True)
+    @patch(
+        "pasqal_cloud.client.Auth0TokenProvider",
+        FakeAuth0AuthenticationSuccess,
+    )
+    def _init_sdk(self):
+        self.sdk = SDK(
+            username="me@test.com",
+            password="password",
+            project_id=str(uuid4()),
+        )
+        self.pulser_sequence = "pulser_test_sequence"
+        self.batch_id = "00000000-0000-0000-0000-000000000001"
+        self.job_id = "00000000-0000-0000-0000-000000022010"
+        self.n_job_runs = 50
+        self.job_variables = {
+            "Omega_max": 14.4,
+            "last_target": "q1",
+            "ts": [200, 500],
+        }
+        self.simple_job_args = {
+            "runs": self.n_job_runs,
+            "variables": self.job_variables,
+        }
+        self.job_result = {"1001": 12, "0110": 35, "1111": 1}
+        self.job_full_result = {
+            "counter": {"1001": 12, "0110": 35, "1111": 1},
+            "raw": ["1001", "1001", "0110", "1001", "0110"],
+        }
+
+    @pytest.mark.usefixtures("mock_request")
+    def test_get_job(self, job: Job):
+        """
+        Asserting the anticipated object is returned with
+        the mocked values when using SDK methods.
+        """
+        job_requested = self.sdk.get_job(job.id)
```

Comment:
Sometimes wonder what's the point in tests like these, when we mock just to compare a returned, mocked string

Reply:
remainder from the past, but yes it does not bring much value